### DISCUSSION TITLE
Screenshot uploading plugin

### DISCUSSION THREAD

**Discussion by IromTypo** [runelite/runelite#15165] [Ideas]
Hello,

I think I have a cool idea. 

What if we could paste screenshots from our clipboard to RuneLite chat, pasting would upload the image, then add a code (::screenshot [code]) to the chat. The text ::screenshot [code] would then be replaced by a hyperlink in the chat.

We could send gear screenshots or memes through ingame chat then for instance :)

Thank you!

**Comment by Enriath** [comment]
RuneLite currently does not store screenshots that you take (they're just put into a folder on your computer), and while there is an option to upload to Imgur, that doesn't really work due to rate limiting. As a result, we'd have to store the screenshots, and quite frankly that sounds like a terrible idea. It's also not really possible to have a link in chat; widgets just don't work like that.

A dedicated chat system is much better for stuff like that, such as Discord, IRC, TeamSpeak, Matrix, anything really.

**Reply by IromTypo** [reply]
Thanks for the reply, that sucks...

**Reply by IromTypo** [reply]
So how possible would it be to have users host the images locally? Like, could RuneLite implement Apache for instance to act as a server for images? Or work as a simple server.

**Reply by Enriath** [reply]
Having users serve images to other users through the use of an embedded server inside the client sounds like the absolute worst idea ever. There'd be issues of security, as users would be directly connecting to others; issues of the network not allowing access to it, which is quite common on home networks; and still doesn't mitigate the issue of sending large amounts of arbitrary data to other users, which has no guarantee to actually be real data at all.

**Reply by IromTypo** [reply]
:(

Will get back to you if I come up with a new solution! Maybe you can help me think about this? I think this would be really good for the client! :)

**Reply by Enriath** [reply]
I still say the solution is to use a dedicated chat system, such as Discord. Modern chatting systems have embeds, moderation tools, and are designed for such things

**Reply by IromTypo** [reply]
Whaaaaaat if users would have to log in to their own Imgur account? That would remove the issue of rate limiting!

**Reply by Enriath** [reply]
> Whaaaaaat if users would have to log in to their own Imgur account? That would remove the issue of rate limiting!

That's an entirely separate suggestion that's been raised, as it would actually make the upload option for the screenshot plugin work consistently. It still doesn't get around the issues of not having links in chat, and there are still issues of sharing arbitrary images that aren't solved by that.

**Reply by IromTypo** [reply]
Oh, so Jagex doesn't allow RL to add hyperlinks to chat?

**Reply by Enriath** [reply]
> It's also not really possible to have a link in chat; widgets just don't work like that.

I said nothing about Jagex allowing us to do anything, because that has nothing to do with anything here.

**Reply by IromTypo** [reply]
Ohhh, so adding a link to a feature like Chat Commands is impossible? Rip, thanks for your time!
